trpcultivate_phenotypes/tests/src/Kernel/Validators/MetadataInputTest.php

```diff
@@ -0,0 +1,180 @@
+  *
+  * @group trpcultivate_phenotypes
+  * @group validators
+  */
+class MetadataInputTest extends ChadoTestKernelBase {
+  use PhenotypeImporterTestTrait;
+
+  /**
+   * A genus that exists and is configured. 
+   */
+  protected $test_genus;
+
+  /**
+   * Modules to enable.
+   */
+  protected static $modules = [
+    'user',
+    'tripal',
+    'tripal_chado',
+    'trpcultivate_phenotypes'
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Set test environment.
+    \Drupal::state()->set('is_a_test_environment', TRUE);
+
+    // Install module configuration.
+    $this->installConfig(['trpcultivate_phenotypes']);
+
+    // Test Chado database.
+    // Create a test chado instance and then set it in the container for use by our service.
+    $this->connection = $this->createTestSchema(ChadoTestKernelBase::PREPARE_TEST_CHADO);
+    $this->container->set('tripal_chado.database', $this->connection);
+
+    // Set plugin manager service.
+    $this->plugin_manager = \Drupal::service('plugin.manager.trpcultivate_validator');
+
+    $genus = 'Tripalus';
+    // Create our organism and configure it.
+    $organism_id = $this->connection->insert('1:organism')
+      ->fields([
+        'genus' => $genus,
+        'species' => 'databasica',
+      ])
+      ->execute();
+
+    $this->assertIsNumeric($organism_id, 'We were not able to create an organism for testing (configured).');
+    $this->test_genus['configured'] = $genus;
+    $this->setOntologyConfig($this->test_genus['configured']);
+
+    // Create another organism but create a configuration item
+    // where the cv for trait is set to 0 - not configured.
+    $genus = 'notconfiggenus';
+    $organism_id = $this->connection->insert('1:organism')
+      ->fields([
+        'genus' => $genus,
+        'species' => 'databasica',
+      ])
+    ->execute();
+
+    $this->assertIsNumeric($organism_id, 'We were not able to create an organism for testing (not configured).');
+    $this->test_genus['not-configured'] = $genus;
+    $config = \Drupal::configFactory()->getEditable('trpcultivate_phenotypes.settings')
+      ->set('trpcultivate.phenotypes.ontology.cvdbon.' . $this->test_genus['not-configured'], ['trait' => 0]);
+    
+    // Set terms configuration.
+    $this->setTermConfig();
+  }
+
+  /**
+   * Test genus input.
+   */
+  public function testGenusInput() {
+    // Create a plugin instance for this validator
+    $validator_id = 'trpcultivate_phenotypes_validator_genus_exists';
+    $instance = $this->plugin_manager->createInstance($validator_id);
+    
+    // The validator expects the form values entered into the form elements
+    // through the $form_state.
+    
+    // Create form fields and set value.
+    // This would have been a form that has been submitted.
+    $form_state = new FormState();
+    // A random field.
+    $form_state->setValues(['project' => uniqid()]);
```

Comment:
It would be better for these tests not to mimic the form submission process so closely. For example, do not create a form state here and populate it. Instead for valid case create the array directly and for invalid case pass in a standard object and a string.

Reply:
Could you expand more. Based on the design, the values are from field element and I just want to make sure that the test captures that extra step of the $form_state getter.

Reply:
Yes, I hear you but that also makes the form state being used first as a potential dependency. Maybe we need to re-word the validator doc block but I meant the note about the form state to be used as hint to the developer but not a requirement. For example, in the future, Tripal Importers may have a drush interface to allow them to be run. In which case, we would want these validators to be useful with a generic array passed in. That is why I made the parameter an array and not just the form state itself --to remove the form state as a requirement.